### PR TITLE
Refatoramento & Retestes

### PR DESCRIPTION
**Pull request by CSRafaelAlves**
Adicionei arquivos ci_android_main.sh e ci_ios_main.sh que passa a deixar no repositório dos desenvolvedores os principais comandos utilizados em CI.
Exibe relatório consolidado para múltiplos devices.
Modifiquei o "run_test_all_devices.sh" para suportarem a possibilidade de retestes.

### CODE REVIEW

File: lib/skeleton/config/scripts/android/run_tests_all_devices.sh [hunk 1]
```diff
@@ -5,10 +5,16 @@
 #
 # $1 -> parameter with the name of the apk
 # $2 -> parameter to indicates the tapume to run. Can be null and can have other 2 values: must or should
+# $3 -> parameter with tests specs. For exxameple, put "feature/<feature_name>" to select an specific feature file; put "" for everything
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/android/run_tests_all_devices.sh:8]: Typo in exxameple (example)

File: lib/skeleton/config/scripts/android/run_tests_all_devices.sh [hunk 2]
```diff
@@ -5,10 +5,16 @@
...
 
 ## CODE BEGIN  #############################################################
 [ $# -lt 1 ] && echo "Wrong number of parameters." && exit 1
 
+#setting which enviroment we are dealing wiff (CI or local)
+enviroment='CI'
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/android/run_tests_all_devices.sh:14]: Can you explain more about this variable? I can't figure out its purpose. Is it the ambiente variable below?
Typo in enviroment (environment)
I think it is a good idea to chance this variable from a local variable to a system variable. Because if you commit this file with environment='local' you will break all your jenkins builds.

File: lib/skeleton/config/scripts/android/run_tests_all_devices.sh [hunk 3]
```diff
@@ -21,21 +27,103 @@ echo Inicio da execução: $(date)
...
+  port=$((port+1)) #increase port
   {
-      ADB_DEVICE_ARG=$device SCREENSHOT_PATH="$reports_path"/"$device"/ calabash-android run $1 -p android -f 'Calabash::Formatters::Html' -o "$reports_path"/"$device"/reports.html -f junit -o "$reports_path"/"$device"
+
+
+     is_emulator=$(echo $device | grep "emulator")
+      if test -n "$is_emulator"
+      then
+        
+        echo Tentativa de destravar a tela de $device
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/android/run_tests_all_devices.sh:72]: Please change all comments and messages to english

File: lib/skeleton/config/scripts/gera_reports_html.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/gera_reports_html.sh:1]: Please change this file name and all variables and comments to english.

File: lib/skeleton/config/scripts/ios/build_app.rb [hunk 5]
```diff
@@ -44,7 +44,7 @@
 
 # Choosing the SDK for device or simulator
 sdk = ''
-if ARGV[1] == 'device'
+if (ARGV[1] == 'devices' || ARGV[1] == 'device')
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/ios/build_app.rb:47]: I see no point in this.
Moreover the help text shows 'device' and you didn't updated it, so no one will know about 'devices' without opening the file.

File: lib/skeleton/config/scripts/ios/ci_ios_main.sh [hunk 6]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Input variables:
+# $1 - The parameters of what test to execute. Example "- @dev" or "" for nothing
+
+ambiente="CI"
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/ios/ci_ios_main.sh:6]: Please change all comments and messages to english

File: lib/skeleton/config/scripts/ios/ci_ios_main.sh [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+then
+	export IOS_WORKSPACE=$WORKSPACE/../isp-participantemobile-ios-master
+	ruby ./config/scripts/ios/build_app.rb jenkins simulator
+	ruby ./config/scripts/ios/build_app.rb jenkins devices
+fi
+export LANG=en_US.UTF-8
+export LANGUAGE=en_US:en
+
+
+#sh $WORKSPACE/config/scripts/ios/run_tests_all_devices.sh ./app/build/ios/simulator/Gebsa-cal.app ./app/build/ios/simulator/Gebsa-cal.app
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/ios/ci_ios_main.sh:20]: Is this really necessary?

File: lib/skeleton/config/scripts/ios/ci_ios_main.sh [hunk 8]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Input variables:
+# $1 - The parameters of what test to execute. Example "- @dev" or "" for nothing
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/ios/ci_ios_main.sh:4]: What test or what tag? Please update this comment to avoid misconception.

File: lib/skeleton/config/scripts/ios/ci_ios_main.sh [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+	export IOS_WORKSPACE=$WORKSPACE/../isp-participantemobile-ios-master
+	ruby ./config/scripts/ios/build_app.rb jenkins simulator
+	ruby ./config/scripts/ios/build_app.rb jenkins devices
+fi
+export LANG=en_US.UTF-8
+export LANGUAGE=en_US:en
+
+
+#sh $WORKSPACE/config/scripts/ios/run_tests_all_devices.sh ./app/build/ios/simulator/Gebsa-cal.app ./app/build/ios/simulator/Gebsa-cal.app
+sh $WORKSPACE/config/scripts/ios/run_tests_all_devices.sh null null "$1"
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/ios/ci_ios_main.sh:21]: Are you sure the first two parameters must be null?
You could export the APP_BUNDLE_PATH variable created in the build_app.rb and save it to use it here.

File: lib/skeleton/config/scripts/ios/ci_ios_main.sh [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+sed -i.bak "s/_STATUS_TITLE_/${STATUS_TITLE}/g" ./config/email/template.html
+sed -i.bak "s/_STATUS_/${STATUS}/g" ./config/email/template.html
+sed -i.bak "s/_JOB_NAME_/${JOB_NAME}/g" ./config/email/template.html
+sed -i.bak "s/_BUILD_NUMBER_/${BUILD_NUMBER}/g" ./config/email/template.html
+sed -i.bak "s/_OS_/Android/g" ./config/email/template.html
+
+
+if [ $ambiente == "CI" ] 
+then
+# Sending the email
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/ios/ci_ios_main.sh:54]: Is this a comment or a command?
Please update this comment to remove all specific uses, like the title of the email

File: lib/skeleton/config/scripts/ios/devices_local.txt [hunk 11]
```diff
@@ -0,0 +1,4 @@
+# <DEVICE_TARGET (UUID)> | <DEVICE_ENDPOINT (IP)> | <DEVICE_NAME> | <DEVICE_TYPE> (Simulator or Device)
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/ios/devices_local.txt:1]: Is this file really necessary? A column in the devices.txt could solve your problem?

File: lib/skeleton/config/scripts/ios/run_tests_all_devices.sh [hunk 12]
```diff
@@ -5,11 +5,16 @@
...
 # $2 -> parameter with the path of the .app bundle for devices
+# $3 -> parameter with the tests to be executed. For example: put "-t @test" or just put "" for everything
+
 
 ## CODE BEGIN  #############################################################
 
 echo Start: $(date)
 
+#sets the enviroment (CI or local)
+ambiente="CI"
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/ios/run_tests_all_devices.sh:16]: Please change all comments and messages to english

File: lib/skeleton/config/scripts/ios/run_tests_all_devices.sh [hunk 13]
```diff
@@ -27,15 +32,79 @@ cd "$original_path"
 cd "$2"
 DEVICE_APP_PATH="$(pwd)"
 
+# showing the versions of calabash-ios in the machine and in the app to be tested:
+echo "A versão do calabash nesta máquina é:"
+calabash-ios version
+echo "A versão do ios-deploy é"
+ios-deploy -V
```
Comment (CSOscarTanner) [inline lib/skeleton/config/scripts/ios/run_tests_all_devices.sh:39]: ios-deploy is not used anymore

**Comment by CSOscarTanner** [comment]
I'm closing this pull request because of inactivity

**Comment by CSRafaelAlves** [comment]
Ok, I'll pull a better improvement later.
